cockatrice/src/window_main.cpp

```diff
@@ -296,6 +297,9 @@ void MainWindow::serverTimeout()
 void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32 endTime)
 {
     switch (r) {
+        case Response::RespClientUpdateRequired:
+            QMessageBox::critical(this, tr("Error"), tr("Your client does not contain features that the server requires, please update your client and try again."));
```

Comment:
Only thing not pictured in that snapshot is the first line above the first missing feature now shows "Missing Features: " instead of being blank, and then each feature is listed on its own line.